lib/screens/translation_screen.dart

```diff
@@ -0,0 +1,360 @@
+    }
+    String fileId = _currentSentence.id;
+    new File("$recordStorage/$fileId.$fileExtension").create().then((filePath) {
+      _flutterSound.startRecorder(filePath.path).then((path) {
+        setState(() {
+          this._isRecording = true;
+        });
+
+        _recorderSubscription =
+            _flutterSound.onRecorderStateChanged.listen((e) {
+          if (e == null) {
+            return;
+          }
+          DateTime date = new DateTime.fromMillisecondsSinceEpoch(
+              e.currentPosition.toInt());
+          String txt = DateFormat('mm:ss:SS', 'en_US').format(date);
+
+          setState(() {
+            this._recorderTimer = txt.substring(0, 8);
+          });
+        });
+      }).catchError((error) {
+        print(error);
+      });
+    }); //File
+  }
+
+  void _stopRecording() {
+    _flutterSound.stopRecorder().then((value) {
+      if (_recorderSubscription != null) {
+        _recorderSubscription.cancel();
+        _recorderSubscription = null;
+      }
+
+      setState(() {
+        this._isRecording = false;
+      });
+    }).catchError((error) {
+      _showMessage(recorderCantStop);
+    });
+
+    _askToSaveRecording();
+  }
+
+  void _doneRecording() {
+    final recordedFile = "${_currentSentence.id}.$fileExtension";
+
+    Translation translation = Translation(
+        author: _author,
+        targetLanguage: _targetLang,
+        sentenceId: _currentSentence.id,
+        audioFileName: recordedFile);
+
+    readFileContentAndUploadTranslation(translation);
+
+    _navigateToNextSentencePage();
+  }
+
+  void _cancelRecording({BuildContext context}) {
+    String fileId = _currentSentence.id;
+    final recordedFile = File("$recordStorage/$fileId.$fileExtension");
+
+    recordedFile.delete().then((result) {
+      setState(() {
+        _recorderTimer = "";
+        _isRecording = false;
+      });
+
+      _showMessage(deletionSuccess, context: context);
+    }).catchError((err) {
+      _showMessage(deletionError, context: context);
+    });
+  }
+
+  void _showMessage(String msg, {BuildContext context}) {
+    Flushbar(
+        message: msg,
+        duration: Duration(seconds: 3),
+        margin: EdgeInsets.all(8),
+        borderRadius: 8,
+        flushbarStyle: FlushbarStyle.FLOATING,
+        icon: Icon(
+          Icons.info_outline,
+          size: 28.0,
+          color: Theme.of(context).primaryColorDark,
+        ),
+        leftBarIndicatorColor: Theme.of(context).primaryColorDark)
+      ..show(context);
+  }
+
+  void _askToSaveRecording() {
+    showDialog<void>(
+        context: context,
+        barrierDismissible: false,
+        builder: (BuildContext context) {
+          return AlertDialog(
+              titleTextStyle: TextStyle(
+                  color: Theme.of(context).primaryColorDark,
+                  fontWeight: FontWeight.bold,
+                  fontSize: 20.0),
+              title: Text("Save Recording"),
```

Comment:
Think about internationalizing this text.

Reply:
yes!